cs/src/comm/interfaces/Listener.cs

```diff
@@ -17,6 +17,14 @@ public class ConnectedEventArgs : EventArgs
         public readonly Connection Connection;
 
         /// <summary>
+        /// A Bond-generated unique identifier for this connection. If you need
+        /// to return an <see cref="Error"/>, give the client this identifier,
+        /// and you will be able to cross-reference it with Bond-generated log
+        /// messages.
+        /// </summary>
+        public string ConnectionId;
```

Comment:
Done. This had a few other effects - check out the second commit I'll push shortly.